packages/utils/src/rundown-utils/rundownUtils.ts

```diff
@@ -1,7 +1,100 @@
-import { OntimeEvent, OntimeRundown } from 'ontime-types';
+import { OntimeEvent, OntimeRundown, OntimeRundownEntry, SupportedEvent } from 'ontime-types';
 
 import { dayInMs } from '../timeConstants.js';
 
+/**
+ * Gets first event in rundown, if it exists
+ * @param {OntimeRundownEntry[]} rundown
+ * @return {OntimeRundownEntry | null}
+ */
+export function getFirst(rundown: OntimeRundownEntry[]) {
+  return rundown.length ? rundown[0] : null;
+}
+
+/**
+ * Gets first scheduled event in rundown, if it exists
+ * @param {OntimeRundownEntry[]} rundown
+ * @return {OntimeEvent | null}
+ */
+export function getFirstEvent(rundown: OntimeRundownEntry[]) {
+  for (let i = 0; i < rundown.length; i++) {
+    if (rundown[i].type === SupportedEvent.Event) {
+      return rundown[i] as OntimeEvent;
+    }
+  }
+  return null;
+}
+
+/**
+ * Gets next event in rundown, if it exists
+ * @param {OntimeRundownEntry[]} rundown
+ * @param {string} currentId
+ * @return {OntimeRundownEntry | null}
+ */
+export function getNext(rundown: OntimeRundownEntry[], currentId: string): OntimeRundownEntry | null {
+  const index = rundown.findIndex((event) => event.id === currentId);
+  if (index !== -1 && index + 1 < rundown.length) {
+    return rundown[index + 1];
+  } else {
+    return null;
+  }
+}
+
+/**
+ * Gets next scheduled event in rundown, if it exists
+ * @param {OntimeRundownEntry[]} rundown
+ * @param {string} currentId
+ * @return {OntimeEvent | null}
+ */
+export function getNextEvent(rundown: OntimeRundownEntry[], currentId: string): OntimeEvent | null {
+  const index = rundown.findIndex((event) => event.id === currentId);
+  if (index < 0) {
+    return null;
+  }
+
+  for (let i = index + 1; i < rundown.length; i++) {
+    if (rundown[i].type === SupportedEvent.Event) {
+      return rundown[i] as OntimeEvent;
+    }
+  }
```

Comment:
Same comment here about using `.find`

Reply:
There is the same answer to all of these that you correctly suggest the use of `.find`. Starting from here:
In this case, I need to start in a specific place and work my way forward to find the next

To use the `.find`, I would need to create an intermediate copy of the array that only contains the events forwards
like `rundown.slice(index - 1).find()`, then do the same for the backwards iterations like `rundown.slice(0, index).find()`

I just thought that the iteration was more explicit and less error prone

Also, since there was an iteration here, I thought having an iteration everywhere would help with predictability and future maintenance

What do you think?

Reply:
I see. I didn't realize it did that. Is my first suggestion of using `.find` valid, though?

Reply:
Absolutely, we can still use .find in one of the cases. But would it not be nicer to have a similar logic in all cases? ie: when requirements change the code that achieves similar things has a similar structure?

As I can see, the change is only semantic

Reply:
Fair enough. Consistency is good 💪🏻